File: Rentify.Services/Database/Property.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Rentify.Services.Database
{
    public class Property
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey(nameof(UserId))]
        public int UserId { get; set; } 
        public User User { get; set; }
        public string Name { get; set; }
        public string Location { get; set; }
        public string City { get; set; }
        public double PricePerDay { get; set; }
        public double PricePerMonth { get; set; }
        public List<string> Tags { get; set; }
        public string NumberOfsquares { get; set; }
        public string Details { get; set; }
        public bool IsAvailable { get; set; }
        public bool IsRentingPerDay {get; set;}
        public bool IsActiveOnApp { get; set; }
    }
}

File: UI/rentify_desktop/lib/models/property.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:rentify_desktop/models/user.dart';

part 'property.g.dart';

@JsonSerializable()
class Property {
  final int id;
  final int userId;
  User? user;
  final String name;
  final String location;
  final String city;
  final double pricePerDay;
  final double pricePerMonth;
  final String numberOfsquares;
  List<String>? tags;
  final String details;
  final bool isAvailable;
  final bool isRentingPerDay;
  final bool isActiveOnApp;

  Property({
    required this.id,
    required this.userId,
    this.user,
    required this.name,
    required this.location,
    required this.city,
    required this.pricePerDay,
    required this.pricePerMonth,
    required this.numberOfsquares,
    this.tags,
    required this.details,
    required this.isAvailable,
    required this.isRentingPerDay,
    required this.isActiveOnApp,
  });

  factory Property.fromJson(Map<String, dynamic> json) =>
      _$PropertyFromJson(json);

  Map<String, dynamic> toJson() => _$PropertyToJson(this);
}
